lib/srv/db/proxyserver.go

```diff
@@ -270,6 +288,57 @@ func (s *ProxyServer) Proxy(ctx context.Context, clientConn, serviceConn io.Read
 	return trace.NewAggregate(errs...)
 }
 
+func (s *ProxyServer) monitorConn(ctx context.Context, authContext *auth.Context, conn net.Conn) (net.Conn, error) {
```

Comment:
Nit: Add a godoc please explaining what this function returns (tracking connection that will be auto-terminated in case disconnect_expired_cert or idle timeout are configured, and unmodified client connection otherwise).